# Big Data Processes - exercise no.2:
## <font color= green> Importing, exploring and visualising data </font> 

### 1. Import and load


Import pandas and load the dataset

In [1]:
import pandas as pd

In [2]:
#we import a new dataset to work with
dataset = pd.read_csv('Auto-miss.csv', delimiter=';')

In [3]:
dataset

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
0        1  18          8          307       130.0  3504.0           12    70   
1        2  15          8          350       165.0  3693.0         11,5    70   
2        3  18          8          318         NaN  3436.0           11    70   
3        4  16          8          304       150.0     NaN           12    70   
4        5  17          8          302       140.0     NaN         10,5    70   
..     ...  ..        ...          ...         ...     ...          ...   ...   
387    393  27          4          140        86.0  2790.0         15,6    82   
388    394  44          4           97        52.0  2130.0         24,6    82   
389    395  32          4          135        84.0  2295.0         11,6    82   
390    396  28          4          120        79.0  2625.0         18,6    82   
391    397  31          4          119        82.0  2720.0         19,4    82   

     origin                       name  
0         1  chevrolet chevelle malibu  
1         1          buick skylark 320  
2         1         plymouth satellite  
3         1              amc rebel sst  
4         1                ford torino  
..      ...                        ...  
387       1            ford mustang gl  
388       2                  vw pickup  
389       1              dodge rampage  
390       1                ford ranger  
391       1                 chevy s-10  

[392 rows x 10 columns]

### 2. Open the dataset and examine it

Using the .head() and .tail() functions to examine the dataset (like we saw last time)

In [4]:
#the .head() function shows the five first rows of the dataset. You can change number of rows shown by inserting a number
#in the parentheses
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 392 entries, 0 to 391
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   obsNo         392 non-null    int64  
 1   mpg           392 non-null    object 
 2   cylinders     392 non-null    int64  
 3   displacement  392 non-null    object 
 4   horsepower    386 non-null    float64
 5   weight        385 non-null    float64
 6   acceleration  392 non-null    object 
 7   year          392 non-null    int64  
 8   origin        392 non-null    int64  
 9   name          392 non-null    object 
dtypes: float64(2), int64(4), object(4)
memory usage: 30.8+ KB


In [5]:
#the .tail() function shows the five last rows of the dataset.
dataset.tail()

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
387    393  27          4          140        86.0  2790.0         15,6    82   
388    394  44          4           97        52.0  2130.0         24,6    82   
389    395  32          4          135        84.0  2295.0         11,6    82   
390    396  28          4          120        79.0  2625.0         18,6    82   
391    397  31          4          119        82.0  2720.0         19,4    82   

     origin             name  
387       1  ford mustang gl  
388       2        vw pickup  
389       1    dodge rampage  
390       1      ford ranger  
391       1       chevy s-10

We can get information on the datatype in each column through the .dtypes function. 

In [6]:
dataset.dtypes

obsNo             int64
mpg              object
cylinders         int64
displacement     object
horsepower      float64
weight          float64
acceleration     object
year              int64
origin            int64
name             object
dtype: object

Information on the column names through .columns

In [7]:
dataset.columns

Index(['obsNo', 'mpg', 'cylinders', 'displacement', 'horsepower', 'weight',
       'acceleration', 'year', 'origin', 'name'],
      dtype='object')

You can get this information and more through the **.info()** function

In [8]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 392 entries, 0 to 391
Data columns (total 10 columns):
obsNo           392 non-null int64
mpg             392 non-null object
cylinders       392 non-null int64
displacement    392 non-null object
horsepower      386 non-null float64
weight          385 non-null float64
acceleration    392 non-null object
year            392 non-null int64
origin          392 non-null int64
name            392 non-null object
dtypes: float64(2), int64(4), object(4)
memory usage: 30.8+ KB


What important information can you see in the **.info()** matrix above:
- range length and index
- columns: name, count, null, type
- datatypes list - count per type
- What columns are missing data

### 3. Exploration with summary statistics

We can use the **.describe()** function to gain basic descriptive statistics about each column where it is applicable. 
For the **.describe()** function the 50% represents the median. However, just because it is applicable for some of the columns does not mean that it creates useful results for all the columns where it as applicable

In [9]:
dataset.describe()

obsNo   cylinders  horsepower       weight        year      origin
count  392.000000  392.000000  386.000000   385.000000  392.000000  392.000000
mean   198.520408    5.471939  104.329016  2973.420779   75.979592    1.576531
std    114.438067    1.705783   38.095547   850.538689    3.683737    0.805518
min      1.000000    3.000000   46.000000  1613.000000   70.000000    1.000000
25%     99.750000    4.000000   75.250000  2223.000000   73.000000    1.000000
50%    198.500000    4.000000   93.500000  2800.000000   76.000000    1.000000
75%    296.250000    8.000000  125.000000  3613.000000   79.000000    2.000000
max    397.000000    8.000000  230.000000  5140.000000   82.000000    3.000000

### 4. Missing data

Not all datasets are complete and contains missing values. It is important to locate any missing values in a dataset in order to later handle them

In [10]:
#we first count the values, which can be counted
dataset.count()

obsNo           392
mpg             392
cylinders       392
displacement    392
horsepower      386
weight          385
acceleration    392
year            392
origin          392
name            392
dtype: int64

The **.isna()** (Is NA?) function can be used to locate the missing values. It returns a boolean (True or False) for each datapoint in the dataset

In [11]:
#All the places where it says True in the dataset, then that is missing data values. 
dataset.isna()

obsNo    mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    False  False      False         False       False   False         False   
1    False  False      False         False       False   False         False   
2    False  False      False         False        True   False         False   
3    False  False      False         False       False    True         False   
4    False  False      False         False       False    True         False   
..     ...    ...        ...           ...         ...     ...           ...   
387  False  False      False         False       False   False         False   
388  False  False      False         False       False   False         False   
389  False  False      False         False       False   False         False   
390  False  False      False         False       False   False         False   
391  False  False      False         False       False   False         False   

      year  origin   name  
0    False   False  False  
1    False   False  False  
2    False   False  False  
3    False   False  False  
4    False   False  False  
..     ...     ...    ...  
387  False   False  False  
388  False   False  False  
389  False   False  False  
390  False   False  False  
391  False   False  False  

[392 rows x 10 columns]

We will learn how to handle missing data in week 4 when we focus on data cleaning

### 5. Filtering (specific rows)

Looking at all the cars is maybe too much so we can focus on one specific car in the dataset

dataset['column name'] returns only the specific column of the dataset 

In [12]:
dataset['name']

0      chevrolet chevelle malibu
1              buick skylark 320
2             plymouth satellite
3                  amc rebel sst
4                    ford torino
                 ...            
387              ford mustang gl
388                    vw pickup
389                dodge rampage
390                  ford ranger
391                   chevy s-10
Name: name, Length: 392, dtype: object

In [13]:
#this is how to select rows based on value
dataset.loc[dataset['name'] == 'vw pickup']

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
388    394  44          4           97        52.0  2130.0         24,6    82   

     origin       name  
388       2  vw pickup

Or all the cars with ford in the name

In [14]:
dataset.loc[dataset['name'].str.contains(r'ford')]

obsNo   mpg  cylinders displacement  horsepower  weight acceleration  \
4        5    17          8          302       140.0     NaN         10,5   
5        6    15          8          429       198.0  4341.0           10   
17      18    21          6          200        85.0  2587.0           16   
25      26    10          8          360       215.0  4615.0           14   
35      37    19          6          250        88.0  3302.0         15,5   
39      41    14          8          351       153.0  4154.0         13,5   
42      44    13          8          400       170.0  4746.0           12   
47      49    18          6          250        88.0  3139.0         14,5   
60      62    21          4          122        86.0  2226.0         16,5   
64      66    14          8          351       153.0  4129.0           13   
73      75    13          8          302       140.0  4294.0           16   
79      81    22          4          122        86.0  2395.0           16   
87      89    14          8          302       137.0  4042.0         14,5   
91      93    13          8          351       158.0  4363.0           13   
99     101    18          6          250        88.0  3021.0         16,5   
103    105    12          8          400       167.0  4906.0         12,5   
111    113    19          4          122        85.0  2310.0         18,5   
128    131    26          4          122        80.0  2451.0         16,5   
134    137    16          8          302       140.0  4141.0           14   
137    140    14          8          302       140.0  4638.0           16   
153    156    15          6          250        72.0  3158.0         19,5   
157    160    14          8          351       148.0  4657.0         13,5   
164    167    13          8          302       129.0  3169.0           12   
166    169    23          4          140        83.0  2639.0           17   
172    175    18          6          171        97.0  2984.0         14,5   
188    191  14,5          8          351       152.0  4215.0         12,8   
191    194    24          6          200        81.0  3012.0         17,6   
198    201    18          6          250        78.0  3574.0           21   
204    207  26,5          4          140        72.0  2565.0         13,6   
212    215    13          8          302       130.0  3870.0           15   
226    229  18,5          6          250        98.0  3525.0           19   
230    233    16          8          351       149.0  4335.0         14,5   
234    237  25,5          4          140        89.0  2755.0         15,8   
243    246  36,1          4           98        66.0  1800.0         14,4   
252    255  20,2          6          200        85.0  2965.0         15,8   
253    256  25,1          4          140        88.0  2720.0         15,4   
262    265  18,1          8          302       139.0  3205.0         11,2   
280    283  22,3          4          140        88.0  2890.0         17,3   
284    287  17,6          8          302       129.0  3725.0         13,4   
288    291  15,5          8          351       142.0  4054.0         14,3   
312    315  26,4          4          140        88.0  2870.0         18,1   
347    352  34,4          4           98        65.0  2045.0         16,2   
348    353  29,9          4           98        65.0  2380.0         20,7   
360    366  20,2          6          200        88.0  3060.0         17,1   
368    374    24          4          140        92.0  2865.0         16,4   
383    389    22          6          232       112.0  2835.0         14,7   
387    393    27          4          140        86.0  2790.0         15,6   
390    396    28          4          120        79.0  2625.0         18,6   

     year  origin                      name  
4      70       1               ford torino  
5      70       1          ford galaxie 500  
17     70       1             ford maverick  
25     70       1                 ford f250  
3

You can also filter based on the row number and not on any name.

In [15]:
#here we want to find out what is in row 48 of the dataset
dataset.iloc[[48]]

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
48     50  23          4          122        86.0  2220.0           14    71   

    origin                name  
48       1  mercury capri 2000

### 6. Combining datasets

Sometimes, the datasets we find do not contain enough information for our project. Consequently, we need to merge datasets in order to construct a dataset which contains all the necessary information.

In [16]:
#we import dataset containing information about the colour of the cars
colour = pd.read_csv('car-colour.csv', delimiter=';')

In [17]:
colour

obsNo             Colour
0        1              Green
1        2              Green
2        3     Blue and green
3        4                Red
4        5     Red and yellow
..     ...                ...
387    393    Black and white
388    394  Purple and bronze
389    395             Bronze
390    396             Bronze
391    397             Silver

[392 rows x 2 columns]

In [18]:
carcolour= pd.merge(dataset, colour, on=['obsNo'], how='outer')

In [19]:
carcolour

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
0        1  18          8          307       130.0  3504.0           12    70   
1        2  15          8          350       165.0  3693.0         11,5    70   
2        3  18          8          318         NaN  3436.0           11    70   
3        4  16          8          304       150.0     NaN           12    70   
4        5  17          8          302       140.0     NaN         10,5    70   
..     ...  ..        ...          ...         ...     ...          ...   ...   
387    393  27          4          140        86.0  2790.0         15,6    82   
388    394  44          4           97        52.0  2130.0         24,6    82   
389    395  32          4          135        84.0  2295.0         11,6    82   
390    396  28          4          120        79.0  2625.0         18,6    82   
391    397  31          4          119        82.0  2720.0         19,4    82   

     origin                       name             Colour  
0         1  chevrolet chevelle malibu              Green  
1         1          buick skylark 320              Green  
2         1         plymouth satellite     Blue and green  
3         1              amc rebel sst                Red  
4         1                ford torino     Red and yellow  
..      ...                        ...                ...  
387       1            ford mustang gl    Black and white  
388       2                  vw pickup  Purple and bronze  
389       1              dodge rampage             Bronze  
390       1                ford ranger             Bronze  
391       1                 chevy s-10             Silver  

[392 rows x 11 columns]

Another way to combine two datasets is the **.concat()**

In [20]:
cardata=[dataset,colour]
carcolour2=pd.concat(cardata, sort=False, join='outer')

In [21]:
carcolour2

obsNo  mpg  cylinders displacement  horsepower  weight acceleration  \
0        1   18        8.0          307       130.0  3504.0           12   
1        2   15        8.0          350       165.0  3693.0         11,5   
2        3   18        8.0          318         NaN  3436.0           11   
3        4   16        8.0          304       150.0     NaN           12   
4        5   17        8.0          302       140.0     NaN         10,5   
..     ...  ...        ...          ...         ...     ...          ...   
387    393  NaN        NaN          NaN         NaN     NaN          NaN   
388    394  NaN        NaN          NaN         NaN     NaN          NaN   
389    395  NaN        NaN          NaN         NaN     NaN          NaN   
390    396  NaN        NaN          NaN         NaN     NaN          NaN   
391    397  NaN        NaN          NaN         NaN     NaN          NaN   

     year  origin                       name             Colour  
0    70.0     1.0  chevrolet chevelle malibu                NaN  
1    70.0     1.0          buick skylark 320                NaN  
2    70.0     1.0         plymouth satellite                NaN  
3    70.0     1.0              amc rebel sst                NaN  
4    70.0     1.0                ford torino                NaN  
..    ...     ...                        ...                ...  
387   NaN     NaN                        NaN    Black and white  
388   NaN     NaN                        NaN  Purple and bronze  
389   NaN     NaN                        NaN             Bronze  
390   NaN     NaN                        NaN             Bronze  
391   NaN     NaN                        NaN             Silver  

[784 rows x 11 columns]

##### Question 1: 
How has concat combined the two datasets in comparison to merge?

You can also combine two datasets by using **.join()**

In [22]:
dataset.join(colour, on='obsNo',rsuffix="_r")

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
0        1  18          8          307       130.0  3504.0           12    70   
1        2  15          8          350       165.0  3693.0         11,5    70   
2        3  18          8          318         NaN  3436.0           11    70   
3        4  16          8          304       150.0     NaN           12    70   
4        5  17          8          302       140.0     NaN         10,5    70   
..     ...  ..        ...          ...         ...     ...          ...   ...   
387    393  27          4          140        86.0  2790.0         15,6    82   
388    394  44          4           97        52.0  2130.0         24,6    82   
389    395  32          4          135        84.0  2295.0         11,6    82   
390    396  28          4          120        79.0  2625.0         18,6    82   
391    397  31          4          119        82.0  2720.0         19,4    82   

     origin                       name  obsNo_r          Colour  
0         1  chevrolet chevelle malibu      2.0           Green  
1         1          buick skylark 320      3.0  Blue and green  
2         1         plymouth satellite      4.0             Red  
3         1              amc rebel sst      5.0  Red and yellow  
4         1                ford torino      6.0  White and grey  
..      ...                        ...      ...             ...  
387       1            ford mustang gl      NaN             NaN  
388       2                  vw pickup      NaN             NaN  
389       1              dodge rampage      NaN             NaN  
390       1                ford ranger      NaN             NaN  
391       1                 chevy s-10      NaN             NaN  

[392 rows x 12 columns]

##### Question 2:
Why do you think that the last 6 rows show NaN for obsNo_r and colour?

In [23]:
#let us look at the rows between row 120 and 130 (130 excluded)
dataset.iloc[120:130]

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
120    122  15          8          318       150.0  3399.0           11    73   
121    123  24          4          121       110.0  2660.0           14    73   
122    124  20          6          156       122.0  2807.0         13,5    73   
123    125  11          8          350       180.0  3664.0           11    73   
124    126  20          6          198        95.0  3102.0         16,5    74   
125    128  19          6          232       100.0  2901.0           16    74   
126    129  15          6          250       100.0  3336.0           17    74   
127    130  31          4           79        67.0  1950.0           19    74   
128    131  26          4          122        80.0  2451.0         16,5    74   
129    132  32          4           71        65.0  1836.0           21    74   

     origin                 name  
120       1    dodge dart custom  
121       2            saab 99le  
122       3       toyota mark ii  
123       1     oldsmobile omega  
124       1      plymouth duster  
125       1           amc hornet  
126       1       chevrolet nova  
127       3          datsun b210  
128       1           ford pinto  
129       3  toyota corolla 1200

Look at obsNo 127? What is wrong? 

#### Exercise 1:

Locate the dataset on who eats the food we grow on Kaggle and load it into a variable called food. Show the first 10 rows of the dataset

In [24]:
#write your code here:
food = pd.read_csv('FAO.csv', delimiter=',', encoding= "ISO-8859-1")
food.head(10)

Area Abbreviation  Area Code         Area  Item Code  \
0               AFG          2  Afghanistan       2511   
1               AFG          2  Afghanistan       2805   
2               AFG          2  Afghanistan       2513   
3               AFG          2  Afghanistan       2513   
4               AFG          2  Afghanistan       2514   
5               AFG          2  Afghanistan       2514   
6               AFG          2  Afghanistan       2517   
7               AFG          2  Afghanistan       2520   
8               AFG          2  Afghanistan       2531   
9               AFG          2  Afghanistan       2536   

                       Item  Element Code Element         Unit  latitude  \
0        Wheat and products          5142    Food  1000 tonnes     33.94   
1  Rice (Milled Equivalent)          5142    Food  1000 tonnes     33.94   
2       Barley and products          5521    Feed  1000 tonnes     33.94   
3       Barley and products          5142    Food  1000 tonnes     33.94   
4        Maize and products          5521    Feed  1000 tonnes     33.94   
5        Maize and products          5142    Food  1000 tonnes     33.94   
6       Millet and products          5142    Food  1000 tonnes     33.94   
7            Cereals, Other          5142    Food  1000 tonnes     33.94   
8     Potatoes and products          5142    Food  1000 tonnes     33.94   
9                Sugar cane          5521    Feed  1000 tonnes     33.94   

   longitude  ...   Y2004   Y2005   Y2006   Y2007   Y2008   Y2009   Y2010  \
0      67.71  ...  3249.0  3486.0  3704.0  4164.0  4252.0  4538.0  4605.0   
1      67.71  ...   419.0   445.0   546.0   455.0   490.0   415.0   442.0   
2      67.71  ...    58.0   236.0   262.0   263.0   230.0   379.0   315.0   
3      67.71  ...   185.0    43.0    44.0    48.0    62.0    55.0    60.0   
4      67.71  ...   120.0   208.0   233.0   249.0   247.0   195.0   178.0   
5      67.71  ...   231.0    67.0    82.0    67.0    69.0    71.0    82.0   
6      67.71  ...    15.0    21.0    11.0    19.0    21.0    18.0    14.0   
7      67.71  ...     2.0     1.0     1.0     0.0     0.0     0.0     0.0   
8      67.71  ...   276.0   294.0   294.0   260.0   242.0   250.0   192.0   
9      67.71  ...    50.0    29.0    61.0    65.0    54.0   114.0    83.0   

    Y2011  Y2012  Y2013  
0  4711.0   4810   4895  
1   476.0    425    422  
2   203.0    367    360  
3    72.0     78     89  
4   191.0    200    200  
5    73.0     77     76  
6    14.0     14     12  
7     0.0      0      0  
8   169.0    196    230  
9    83.0     69     81  

[10 rows x 63 columns]

#### Exercise 2:
What is the data unit in the food dataset? The data unit is the instace that is described by all the variables in the dataset

**answer:** Item - because Item is what is described by all the other variables in the dataset

#### Exercise 3:

Locate the dataset on One-Sided violence "Government of Angola - Civilians" in the UCDP (Uppsala Conflict Data Programme) and load it into a variable called df (alias for dataframe). Find out the datatype in each column and the name of each column 

In [25]:
#write your code here:
df= pd.read_csv('gedevents-2021-02-09.csv', delimiter=',') 

In [26]:
df.dtypes


id                    object
relid                  int64
year                    bool
active_year           object
code_status            int64
type_of_violence       int64
conflict_dset_id       int64
conflict_new_id       object
conflict_name          int64
dyad_dset_id           int64
dyad_new_id           object
dyad_name              int64
side_a_dset_id         int64
side_a_new_id         object
side_a                 int64
side_b_dset_id         int64
side_b_new_id         object
side_b                 int64
number_of_sources     object
source_article        object
source_office         object
source_date           object
source_headline       object
source_original        int64
where_prec            object
where_coordinates     object
where_description     object
adm_1                 object
adm_2                float64
latitude             float64
longitude             object
geom_wkt               int64
priogrid_gid          object
country                int64
country_id    

In [27]:
df.columns

Index(['id', 'relid', 'year', 'active_year', 'code_status', 'type_of_violence',
       'conflict_dset_id', 'conflict_new_id', 'conflict_name', 'dyad_dset_id',
       'dyad_new_id', 'dyad_name', 'side_a_dset_id', 'side_a_new_id', 'side_a',
       'side_b_dset_id', 'side_b_new_id', 'side_b', 'number_of_sources',
       'source_article', 'source_office', 'source_date', 'source_headline',
       'source_original', 'where_prec', 'where_coordinates',
       'where_description', 'adm_1', 'adm_2', 'latitude', 'longitude',
       'geom_wkt', 'priogrid_gid', 'country', 'country_id', 'region',
       'event_clarity', 'date_prec', 'date_start', 'date_end', 'deaths_a',
       'deaths_b', 'deaths_civilians', 'deaths_unknown', 'best_est',
       'high_est', 'low_est'],
      dtype='object')

#### Exercise 4:

How does the UCDP collect their data on conflicts and how may this impact research based on this data? 

**answer:** Go to UCDP: https://ucdp.uu.se/
	- Press on "about"
	- Press "methodology"
	- **Data Collection: the data is based on news reports**
		○ At least one global newswires (like Reuters or AFP) in addition to BBC Monitoring
		○ 50.000 news reports collected which are evaluated by human coders--> 10.000-12.000 events are coded annually because of duplicates
	- Source evaluation, (bias): 
		○ Fatality numbers are based on publicly avaiable sources--> there is a lack of information in conflict zones so that means it is likely that there are more fatalities than what the data says--> the data is best interpreted as a baseline


#### Exercise 5: 

Which additional datasets, reports etc. could prove beneficial for a research project based on the dataset on One-Sided Violence "Government of Angola - Civilians"? Describe how you would use the additional material in a project

**Answer:** 	
- Institute for economics and peace makes every year a global peace index: https://www.economicsandpeace.org/?s=the+global+peace+index
Angola is number 91--> use the data to problematize how the Institute for economics and peace makes the global peace index

#### Exercise 6:

Load the dataset "car-maxspeed.csv" into a variable called max-speed

In [28]:
#write your code here
max_speed=pd.read_csv('car-maxspeed.csv', delimiter=';') 
max_speed

obsNo  max speed (km/hour)
0        1                  180
1        2                  175
2        3                  166
3        4                  230
4        5                  300
..     ...                  ...
387    393                  145
388    394                  176
389    395                  187
390    396                  175
391    397                  134

[392 rows x 2 columns]

#### Exercise 7:

Combine the "carcolour" dataset with the "max-speed" dataset in a new dataset called "carcolourspeed" 

In [29]:
#write your code here

carcolourspeed= pd.merge(carcolour, max_speed, on=['obsNo'], how='outer')
carcolourspeed

obsNo mpg  cylinders displacement  horsepower  weight acceleration  year  \
0        1  18          8          307       130.0  3504.0           12    70   
1        2  15          8          350       165.0  3693.0         11,5    70   
2        3  18          8          318         NaN  3436.0           11    70   
3        4  16          8          304       150.0     NaN           12    70   
4        5  17          8          302       140.0     NaN         10,5    70   
..     ...  ..        ...          ...         ...     ...          ...   ...   
387    393  27          4          140        86.0  2790.0         15,6    82   
388    394  44          4           97        52.0  2130.0         24,6    82   
389    395  32          4          135        84.0  2295.0         11,6    82   
390    396  28          4          120        79.0  2625.0         18,6    82   
391    397  31          4          119        82.0  2720.0         19,4    82   

     origin                       name             Colour  max speed (km/hour)  
0         1  chevrolet chevelle malibu              Green                  180  
1         1          buick skylark 320              Green                  175  
2         1         plymouth satellite     Blue and green                  166  
3         1              amc rebel sst                Red                  230  
4         1                ford torino     Red and yellow                  300  
..      ...                        ...                ...                  ...  
387       1            ford mustang gl    Black and white                  145  
388       2                  vw pickup  Purple and bronze                  176  
389       1              dodge rampage             Bronze                  187  
390       1                ford ranger             Bronze                  175  
391       1                 chevy s-10             Silver                  134  

[392 rows x 12 columns]

#### Exercise 8:

locate all cars with "chevrolet" in the name in the "carcolourspeed" dataset and print them

In [30]:
#write your code here
carcolourspeed.loc[carcolourspeed['name'].str.contains(r'chevrolet')]

obsNo   mpg  cylinders displacement  horsepower  weight acceleration  \
0        1    18          8          307       130.0  3504.0           12   
6        7    14          8          454       220.0  4354.0            9   
12      13    15          8          400       150.0  3761.0          9,5   
30      31    28          4          140        90.0  2264.0         15,5   
34      36    17          6          250       100.0  3329.0         15,5   
37      39    14          8          350       165.0  4209.0           12   
45      47    22          4          140        72.0  2408.0           19   
59      61    20          4          140        90.0  2408.0         19,5   
61      63    13          8          350       165.0  4274.0           12   
72      74    13          8          307       130.0  4098.0           14   
86      88    13          8          350       145.0  3988.0           13   
90      92    13          8          400       150.0  4464.0           12   
97      99    16          6          250       100.0  3278.0           18   
102    104    11          8          400       150.0  4997.0           14   
108    110    21          4          140        72.0  2401.0         19,5   
114    116    15          8          350       145.0  4082.0           13   
126    129    15          6          250       100.0  3336.0           17   
130    133    25          4          140        75.0  2542.0           17   
131    134    16          6          250       100.0  3781.0           17   
151    154    18          6          250       105.0  3459.0           16   
155    158    15          8          350       145.0  4440.0           14   
163    166    20          8          262       110.0  3221.0         13,5   
185    188  17,5          8          305       140.0  4215.0           13   
190    193    22          6          250       105.0  3353.0         14,5   
193    196    29          4           85        52.0  2035.0         22,2   
194    197  24,5          4           98        60.0  2164.0         22,1   
219    222  17,5          8          305       145.0  3880.0         12,5   
223    226  17,5          6          250       110.0  3520.0         16,4   
228    231  15,5          8          350       170.0  4165.0         11,4   
235    238  30,5          4           98        63.0  2051.0           17   
251    254  20,5          6          200        95.0  3155.0         18,2   
260    263  19,2          8          305       145.0  3425.0         13,2   
264    267    30          4           98        68.0  2155.0         16,5   
283    286    17          8          305       130.0  3840.0         15,4   
289    292  19,2          8          267       125.0  3605.0           15   
304    307  28,8          6          173       115.0  2595.0         11,3   
309    312  32,1          4           98        70.0  2120.0         15,5   
311    314    28          4          151        90.0  2678.0         16,5   
337    342  23,5          6          173       110.0  2725.0         12,6   
362    368    28          4          112        88.0  2605.0         19,6   
363    369    27          4          112        88.0  2640.0         18,6   
364    370    34          4          112        88.0  2395.0           18   
386    392    27          4          151        90.0  2950.0         17,3   

     year  origin                               name           Colour  \
0      70       1          chevrolet chevelle malibu            Green   
6      70       1                   chevrolet impala            Black   
12     70       1              chevrolet monte carlo             Gold   
30     71       1                chevrolet vega 2300              Red   
34     71       1          chevrolet chevelle malibu           Orange   
37     71       1                   chevrolet impala             Gold   
45     71       1                chevrolet vega (sw)  Black and white   
59     72       1               

# That's it! Have a great day :-) 